justfile

```diff
@@ -0,0 +1,94 @@
+# recipes for the `just` command runner: https://just.systems
+# how to install: https://github.com/casey/just#packages
+
+# the central source of truth for Bonfire extension project boilerplate
+# is tracked at https://github.com/bonfire-networks/bonfire-extension-boilerplate
+
+# we load all vars from .env file into the env of just commands
+set dotenv-load
+# and export just vars as env vars
+set export
+
+## Main configs - override these using env vars
+
+APP_VSN_EXTRA := env_var_or_default("APP_VSN_EXTRA", "")
+DB_DOCKER_VERSION := env_var_or_default('DB_DOCKER_VERSION', "16-3.4")
+DB_DOCKER_IMAGE := env_var_or_default('DB_DOCKER_IMAGE', if arch() == "aarch64" { "ghcr.io/baosystems/postgis:"+DB_DOCKER_VERSION } else { "postgis/postgis:"+DB_DOCKER_VERSION+"-alpine" })
+DB_STARTUP_TIME := env_var_or_default("DB_STARTUP_TIME", "10")
+POSTGRES_PORT := env_var_or_default("POSTGRES_PORT", "5432")
+export MIX_ENV := env_var_or_default("MIX_ENV", "test")
+export POSTGRES_USER := env_var_or_default("POSTGRES_USER", "postgres")
+export POSTGRES_PASSWORD := env_var_or_default("POSTGRES_PASSWORD", "postgres")
+export POSTGRES_DB := "localhost:" + POSTGRES_PORT
+
+## Configure just
+# choose shell for running recipes
+set shell := ["bash", "-uec"]
+# support args like $1, $2, etc, and $@ for all args
+set positional-arguments
+
+
+#### COMMANDS ####
+
+help:
+    @echo "Just commands:"
+    @just --list
+
+compile:
+    mix compile
+
+clean: stop-test-db clean-symlinks
+    mix deps.clean --all
+    rm -rf .hex .mix .cache _build deps
+
+clean-symlinks:
+    find lib/ -type l -delete
+
+boilerplate-update:
+    mkdir -p .bonfire-extension-boilerplate
+    git clone https://github.com/bonfire-networks/bonfire-extension-boilerplate.git --branch main --single-branch .bonfire-extension-boilerplate
+    cd .bonfire-extension-boilerplate && cp .envrc justfile .gitignore .. && cp .github/workflows/main.yml ../.github/workflows/main.yml && cp lib/mix/mess.ex ../mess.exs
+    rm -rf .bonfire-extension-boilerplate
+
+deps-get:
+    mix deps.get
+
+deps-update:
+    mix deps.update --all
+
+common-mix-tasks-setup:
+    cd lib/mix/ && ([ -d ../../deps/bonfire_common/lib/mix_tasks ] && ln -sf ../../deps/bonfire_common/lib/mix_tasks tasks || ln -sf ../mix_tasks tasks) && cd -
+    cd lib/mix/tasks/release/ && MIX_ENV=prod mix escript.build && cd -
+
+ext-migrations-copy: common-mix-tasks-setup
+    mkdir -p priv/repo
+    mix bonfire.extension.copy_migrations --to priv/repo/migrations --repo Bonfire.Common.Repo --force
```

Comment:
this will probably be replaced by igniter's `install` and `upgrade` but no need to change anything now, luckily will be easy to change with `boilerplate-update` :)